vendor/bat-native-ledger/src/bat/ledger/internal/promotion.cc

```diff
@@ -381,6 +387,10 @@ void Promotion::OnGetAllPromotions(
 
   if (list.size() > 0) {
     for (auto & item : list) {
+      if (item->legacyClaimed) {
+        ClaimTokens(std::move(item), [](const ledger::Result _){});
```

Comment:
we should mark status of this promotion as claimed

Reply:
`item->status = ledger::PromotionStatus::CLAIMED` - yep?

Reply:
@NejcZdovc just asking - wouldn't 'ClaimTokens' do this?

Reply:
it will, the problem is that you are sending this legacy promotions to UI, so we need to marked them asap so that they are not shown to users

Reply:
Done